lib/Backend/IRBuilder.cpp

```diff
@@ -6074,7 +6074,16 @@ IRBuilder::BuildProfiledCallI(Js::OpCode opcode, uint32 offset, Js::RegSlot retu
         newOpcode = opcode;
         Js::OpCodeUtil::ConvertNonCallOpToNonProfiled(newOpcode);
         Assert(newOpcode == Js::OpCode::NewScObject || newOpcode == Js::OpCode::NewScObjectSpread);
-        returnType = ValueType::GetObject(ObjectType::UninitializedObject);
+        if (newOpcode == Js::OpCode::NewScObjectSpread || !this->m_func->HasProfileInfo())
```

Comment:
After a couple of related fixes, NewScObjectSpread profile information use is now enabled in this PR as well.